core/descriptors/src/org/jetbrains/kotlin/types/TypeSubstitution.kt

```diff
@@ -144,23 +146,26 @@ fun KotlinType.replace(
 ): KotlinType {
     if ((newArguments.isEmpty() || newArguments === arguments) && newAnnotations === annotations) return this
 
+    val newAttributes = attributes.replaceAnnotations(
+        // Specially handle FilteredAnnotations here due to FilteredAnnotations.isEmpty()
+        if (newAnnotations is FilteredAnnotations && newAnnotations.isEmpty()) Annotations.EMPTY else newAnnotations
```

Comment:
There is a strange(?) implementation of FilteredAnnotations.isEmpty() that some of the tests rely on: https://github.com/JetBrains/kotlin/blob/master/core/descriptors/src/org/jetbrains/kotlin/descriptors/annotations/Annotations.kt#L71